kore/src/test/java/org/kframework/POSetTest.java

```diff
@@ -80,4 +80,17 @@ public void glb() {
     assertEquals(
         Optional.empty(), new POSet<>(Pair.of(b2, b1), Pair.of(b3, b2), Pair.of(b4, b2)).minimum());
   }
+
+  @Test
+  public void connectedComponents() {
```

Comment:
Consider turning this into a `Parameterized` test with multiple test inputs.

Reply:
Hard to do this as the property being asserted changes depending on the poset structure, but I've added some more tests.

Reply:
This solution is fine.

Alternatively, you can represent the expected result as a partition of the elements (e.g. `Collection<? extends List<? extends T>> expected`) and assert

* `for all X in expected . for all x1 in X . for all x2 in X: id(x1) == id(x2)`
* `for all X in expected . for all Y in expected . X.size() > 0 && Y.size() > 0 &&  id(X.get(0)) != id(Y.get(0))`
*  (optionally) `id` is onto `[0, N)`

Reply:
Ah, very nice - clever setup!

Reply:
Looking at the [docs](https://www.baeldung.com/parameterized-tests-junit-5), doing this concisely needs JUnit 5 and we're currently using 4 so I may stick with the hand-coded examples for now.